fabric/externalBuilder/bin/lib/common.sh

```diff
@@ -68,23 +67,24 @@ cc_build_for_docker() {
 }
 
 # assumes CC_BUILD_DIR & CC_RT_METADATA_DIR
-# provides: SGX_MODE, PEER_ADDRESS, TLS_ARTIFACTS_DIR & all env-vars expected by shim
+# provides: SGX_MODE, PEER_ADDRESS, RUN_STATE_DIR & all env-vars expected by shim
 process_runtime_metadata() {
-    # Note: while the sample scripts re-use ${CC_RT_METADATA_DIR} to extract the
-    # artifacts from chaincode.json, the docu explicitly says one should treat
-    # that dir as read-only. So we create a separate tmp directory to store
-    # extracted artifacts.
-
-    TLS_ARTIFACTS_DIR="/tmp/fpc-extbuilder.$$"
-    try mkdir -p "${TLS_ARTIFACTS_DIR}"
-
-
     [ -f "${CC_BUILD_DIR}/${METADATA_FILE}" ] || die "no metadata file '${METADATA_FILE}'"
     SGX_MODE="$(jq -r .sgx_mode "${CC_BUILD_DIR}/metadata.json")"
     [ ! -z "${SGX_MODE}" ]                       || die "SGX mode not specified in metadata file"
 
     [ -f "${CC_RT_METADATA_DIR}/chaincode.json" ] || die "chaincode.jsaon does not exist"
     export CORE_CHAINCODE_ID_NAME="$(jq -r .chaincode_id "${CC_RT_METADATA_DIR}/chaincode.json")" || die "could not extract chaincode-id"
+
+	
+    SHORT_CHAINCODE_ID_NAME=$(echo ${CORE_CHAINCODE_ID_NAME} | sed 's/\(.*:.\{5\}\).*/\1/')
+    RUN_STATE_DIR="/tmp/fpc-extbuilder.${SHORT_CHAINCODE_ID_NAME}.$$"
```

Comment:
if the cleanup variable is not set, this does not help -- the problem is the pid!
As we discussed, the timestamp should solve this:
```suggestion
    RUN_STATE_DIR="/tmp/fpc-extbuilder.${SHORT_CHAINCODE_ID_NAME}.$(date +%s%N)"
```
(I have not tested the suggestion)

Reply:
actually, you already _do_ have a timestamp with the directory entry (e.g., `ls -ltra /tmp/fpc*` will give you the most recently used one.   I thought you would be fine with that.   This gives though slightly different info than above (least recently touched vs creation time) so both could be useful.  For date format i would suggest though `date '+%Y-%m-%d_%H:%M:%s'` which gives more meaning output.   Also, for sort-order, you want `<name, date>` not `<date,name>` ?

Reply:
I had misunderstood, but yes, I am fine.

FYI: <name, date> is also fine to determine the last one per instance.